app/controllers/books_controller.rb

```diff
@@ -5,7 +5,8 @@ class BooksController < ApplicationController
 
   # GET /books or /books.json
   def index
-    @books = Book.all
+    per = 4
+    @books = Book.page(params[:page]).per(per).order(:created_at, :id)
```

Comment:
```suggestion
    @books = Book.page(params[:page]).per(4).order(:created_at, :id)
```

変数に入れなくてもあまり可読性は変わらない気がします。

あと、特に決まりがあるわけではないですが、kaminariのメソッドはあとに書く方が多いかもしれません。

```suggestion
    @books = Book.order(:created_at, :id).page(params[:page]).per(4)
```

Reply:
コメントありがとうございます。
まだ、メソッドの引数とマジックナンバーの違いがよくわかっていないです。
今回の場合、自分で定義した数字ではなくkaminariのメソッドとして引数が必要なのでそのまま数字を入れても問題ないのでしょうか？

Reply:
> まだ、メソッドの引数とマジックナンバーの違いがよくわかっていないです。
今回の場合、自分で定義した数字ではなくkaminariのメソッドとして引数が必要なのでそのまま数字を入れても問題ないのでしょうか？

なるほど、良い質問ですね！
マジックナンバーは「これ、何の数値？」っていう疑問が出てくるような数値を指します。
疑問がわかないケース（用途が明らかなケース）はマジックナンバーとは呼びません。

```ruby
# 326と561って何？→マジックナンバー
window.change_size(326, 561)

# 変数に入れた方がわかりやすい
width = 326
height = 561
window.change_size(width, height)
```

```ruby
# 561はたぶん画面の高さを指定しているはず→マジックナンバーではない
window.change_height(561)
```

疑問がわく・わかないは主観によるので、人によって意見が分かれるケースもありますが、今回の`per(4)`は「4件ごと」と素直に読めるので、ほとんどの人は疑問に思わないかなーと思います。

Reply:
具体例がとてもわかりやすかったです。メソッドである程度意味が理解できるもの、読んでて意味が分かりづらいものが判断軸なんですね。ありがとうございます！！